src/app/mycart/my-cart.component.ts

```diff
@@ -1,15 +1,82 @@
 import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router} from '@angular/router';
+import { ProductsService } from '../products/products.service';
 
 @Component({
   selector: 'app-my-cart',
   templateUrl: './my-cart.component.html',
   styleUrls: ['./my-cart.component.css']
 })
 export class MyCartComponent implements OnInit {
+  name:string;
+  productsList:any=[];
+  myCart:any=[];
+  cartValues=[];
+  totalAmount:number=0;
 
-  constructor() { }
+  constructor(private router:Router,
+              private route:ActivatedRoute,
+              private productsService:ProductsService)
+              { }
 
   ngOnInit(): void {
+    this.name=this.route.snapshot.paramMap.get('name');
+    this.productsService.getProducts().subscribe(item=>{
+      this.productsList=item;
+
+      this.cartValues=JSON.parse(localStorage.getItem('cartValue')) || [];
+      console.log("first",this.cartValues); 
+
+      for(let i=0;i<this.productsList.length;i++){
+        if(this.productsList[i].name===this.name && 
+          !this.cartValues.map(item=>item.name).includes(this.name)){
+            this.cartValues.push(this.productsList[i]);
+            console.log("after push",this.productsList[i]);
+            localStorage.setItem('cartValue',JSON.stringify(this.cartValues));
+            let cart=JSON.parse(localStorage.getItem('cartValue'));
+            console.log("second",cart);         
+        }
+      }           
+    })
+  }
+
+  productToDelete(value){
+    console.log("item"+value);
+    this.cartValues.splice(this.cartValues.findIndex(item=>item.id==value),1);
```

Comment:
you can just use a filter instead, than using findIndex within splice. Try to keep your code as functional as possible, with minimum mutations. you can read about it here https://medium.com/@cristiansalcescu/learn-immutability-with-javascript-6a67e4a48d7f